lib/elixir/lib/version.ex

```diff
@@ -119,6 +119,27 @@ defmodule Version do
     defstruct [:source, :matchspec, :compiled]
 
     @opaque t :: %__MODULE__{source: String.t(), matchspec: :ets.match_spec(), compiled: boolean}
+
+    @doc false
+    def new(source, spec) do
```

Comment:
I think we need to include `@spec` here to avoid warnings too:

```elixir
@spec new(String.t(), :ets.match_spec()) :: t
@spec compile(t) :: t
@spec match?(t, tuple) :: boolean # or expanding "tuple" to the "{non_neg_integer, non_neg_integer, non_neg_integer, [String.t() | non_neg_integer], boolean}"
```

or expanding `tuple` in the last one to `{non_neg_integer, non_neg_integer, non_neg_integer, [String.t() | non_neg_integer], boolean}`

Reply:
Those won't cause warning because of the success typing :)

But I can surely do that (and would like to)

Reply:
I'll simplify to tuple because otherwise I'd like that 5-tuple to be a type defined and exported at Version.